aiida_quantumespresso/calculations/pw.py

```diff
@@ -152,6 +155,38 @@ def define(cls, spec):
                     'is `False` and/or `electron_maxstep` is 0.')
         # yapf: enable
 
+    @staticmethod
+    def validate_inputs(value, _):
+        """Validate the top level namespace.
+
+        1. Check that the restart input parameters are set correctly. In case of 'nscf' and 'bands' calculations, this
+        means that ``parent_folder`` is provided, ``startingpot`` is set to 'file' and ``restart_mode`` is
+        'from_scratch'. For other calculations, if the ``parent_folder`` is provided, the restart settings must be set
+        to use some of the outputs.
+        """
+        parameters = value['parameters'].get_dict()
+        calculation_type = parameters.get('CONTROL', {}).get('calculation', 'scf')
+
+        # Check that the restart input parameters are set correctly
+        if calculation_type in ('nscf', 'bands'):
+            if 'parent_folder' not in value:
+                return f'`parent_folder` not provided for `{calculation_type}` calculation.'
+            if parameters.get('ELECTRONS', {}).get('startingpot', 'file') != 'file':
+                return f'`startingpot` should be set to `file` for a `{calculation_type}` calculation.'
+            if parameters.get('CONTROL', {}).get('CONTROL', None) == 'restart':
+                return f'`restart_mode` should be set to `from_scratch` for a `{calculation_type}` calculation.'
```

Comment:
Think this should be
```suggestion
                return f'`restart_mode` should be set to `restart` for a `{calculation_type}` calculation.'
```
or otherwise the check doesn't make any sense

Reply:
Actually, I think it does?

With the correct key being requested in the CONTROL namespace:

```Python
            if parameters.get('CONTROL', {}).get('restart_mode', None) == 'restart':
                return f'`restart_mode` should be set to `from_scratch` for a `{calculation_type}` calculation.'
```

It will basically check if the `restart_mode` has been set to `'restart'`.  This may sound intuitively correct, but looking at the [pw.x input description](https://www.quantum-espresso.org/Doc/INPUT_PW.html#idm60):

```
restart' :
From previous interrupted run. Use this switch only if you want to
continue, using the same number of processors and parallelization,
an interrupted calculation. Do not use to start a new one, or to
perform a non-scf calculations.  Works only if the calculation was
cleanly stopped using variable max_seconds, or by user request
with an "exit file" (i.e.: create a file "prefix".EXIT, in directory
"outdir"; see variables prefix, outdir). The default for
startingwfc and startingpot is set to 'file'.
```

So in order to properly start from the charge density (as required by `nscf` and `bands` calculations), you have to either not set this tag, or set it explicitly to `'from_scratch'` (the default).

Reply:
Why not check directly that `parameters.get('CONTROL', {}).get('restart_mode', 'from_scratch') != 'from_scratch'`? It catches more possible problems, no? (i.e. `restart_mode = typo`)

Reply:
Yeah, you are correct, but I misread because I was expecting a negative conditional like @ramirezfranciscof suggests. That makes more sense and also catches other problems if it is accidentally to set any other value

Reply:
Right, makes sense. More readable and functional, love it! 👌

Reply:
Wait, I am a bit confused, for nscf calculation if I only set `restart_mode = restart` without any `startingpot` and `startingwfc`, it should also work, but this validator will invalidate this usage?

Reply:
If `calculation == 'nscf'` it won't get in this part of the conditional though. And even then, according to the docs, `restart_mode = restart` is non-sensical for `nscf` and one should use `from_scratch`.

Reply:
Thanks!
I tried to look into QE code, indeed the `restart_mode = restart` tag is aiming to restart from an unfinished but correctly stopped previous calculation (it could read additional temp files during scf iteration). When the previous calculation successfully finished (and correctly stopped), the `calculation == 'nscf'` will default `startingpot` to `file` and `startingwfc` to `atomic+random`, in this case although the `restart_mode = restart` cannot find the additional temp files during scf iteration, it however does not harm and QE keeps running nscf. 
So my previous nscf are always using `restart_mode = restart`, they work but in theory I should use `restart_mode = from_scratch` together with `startingpot = file`.

Reply:
I see, thanks for the comments @qiaojunfeng! So basically the following validation is in principle not necessary:

```python
            if parameters.get('CONTROL', {}).get('restart_mode', 'from_scratch') != 'from_scratch':
                return f'`restart_mode` should be set to `from_scratch` for a `{calculation_type}` calculation.'
```

Since QE will _only_ restart from the `startingpot` file anyways, even though `restart_mode` is set to `'restart'`? Still, would probably be best to nudge users to just setting the correct input tags according to the documentation.

Reply:
This is what I wrote in an original comment (which I can't find any longer) that before we wouldn't really raise an error if there were "inconsistent" parameters defined, and the workchain would simply set what is correct, even if those inconsistent parameters would work fine in QE because it would ignore certain conflicting values. I can see how there is something to be said for warning users for inconsistent input, but maybe we should not start raising errors like we currently are doing for `nscf` and `bands` calculations.

Reply:
> ```python
>             if parameters.get('CONTROL', {}).get('restart_mode', 'from_scratch') != 'from_scratch':
>                 return f'`restart_mode` should be set to `from_scratch` for a `{calculation_type}` calculation.'
> ```

Yes it is not necessary. For `calculation = 'nscf'`, `restart_mode = 'restart'` also works, and remove `restart_mode` can work as well. Maybe if the user has set a `restart_mode = 'restart'` for nscf, we just print a warning instead of an error?

Reply:
> This is what I wrote in an original comment (which I can't find any longer) that before we wouldn't really raise an error if there were "inconsistent" parameters defined, and the workchain would simply set what is correct, even if those inconsistent parameters would work fine in QE because it would ignore certain conflicting values.

I still will object to silently "fixing" inconsistent inputs on the fly out of principle. 🙃 If the user is providing non-sensible inputs, something is wrong with how they are setting up the calculations, and they should be made aware about that.

I'm happy to raise a warning for the final conditional, since the calculation will still run as intended. The first two will not run, if I understand correctly, so these should raise.

Reply:
> I still will object to silently "fixing" inconsistent inputs on the fly out of principle. upside_down_face If the user is providing non-sensible inputs, something is wrong with how they are setting up the calculations, and they should be made aware about that.

Normally I would agree, but Quantum ESPRESSO does the same thing. For that reason, lot's of people will actually specify `restart_mode = restart` for an NSCF calculation, because that is what makes sense intuitively and it works, because QE just silently ignores it. In this case I think it is probably not a good idea if this starts breaking stuff on our side all of a sudden.

> I'm happy to raise a warning for the final conditional, since the calculation will still run as intended. The first two will not run, if I understand correctly, so these should raise.

Seems like the golden middle road. If you add this final change we can merge.